### DISCUSSION TITLE
KNeighborsTimeSeriesClassifier Performance issues?

### DISCUSSION THREAD

**Discussion by sly8787** [alan-turing-institute/sktime#2714] [Q&A]
Hello folks, can it be that distance-calculation with jit is not used correctly in combination with the classifier of sklearn?  

<img width="1475" alt="Bildschirmfoto 2022-05-28 um 20 50 15" src="https://user-images.githubusercontent.com/179658/170839373-af88a81f-548b-41a7-a6ff-2ae64e127cf9.png">

**Comment by fkiraly** [comment]
FYI @chrisholder

**Comment by fkiraly** [comment]
Maybe related to this problem I just discovered?
https://github.com/alan-turing-institute/sktime/issues/2774

**Reply by sly8787** [reply]
I think this is more due to how the distance is calculated – for example, why is `pairwise_distances_chunked` of `sklearn` used and not the implementation of `sktime` as in it's regression counterpart (`KNeighborsTimeSeriesRegressor`)?

**Reply by fkiraly** [reply]
don't really know - you would have to ask @jasonlines or @TonyBagnall.

I've implemented a refactored version that seems simpler here: https://github.com/alan-turing-institute/sktime/pull/1998 and uses indeed the `sktime` distances directly.

But currently there's some reservations precisely around the internal use of distance that prevent this from being merged.

**Reply by fkiraly** [reply]
perhaps you would like to try the #1998 version and see how it performs in your case?

**Reply by fkiraly** [reply]
just resolved some merge conflicts so you can try if you want
